### PR TITLE
fix(DII-417): added conditional Param support

### PR DESCRIPTION
**Pull request by kridai**
# Fixes #
Closes : DII-417
Changes added for adding new constructors based on conditional params. Changes for adding conditional params is represented by [this](https://github.com/twilio/twilio-oai/pull/59) PR. 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: examples/twilio_api_v2010.yaml [hunk 1]
```diff
@@ -239,6 +239,12 @@ paths:
           description: Created
       security:
         - accountSid_authToken: [ ]
+      x-twilio:
+        conditional:
+        - - TestInteger
```
Comment (childish-sambino) [inline examples/twilio_api_v2010.yaml:244]: Why does this have double dashes?

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 2]
```diff
@@ -96,7 +99,7 @@ private void updateAccountSidParam(final String path, final PathItem pathMap) {
         pathMap.readOperations().stream().map(io.swagger.v3.oas.models.Operation::getParameters)
         .filter(Objects::nonNull)
         .flatMap(Collection::stream)
-        .filter(param -> param.getIn().equals("path") && param.getName().equals("AccountSid"))
+        .filter(param -> param.getIn().equals("path") && ( ACCOUNT_SID_FORMAT.equals(param.getSchema().getPattern())))
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:102]: Why was this change needed?

File: examples/twilio_api_v2010.yaml (hunk 1 above)
Comment (kridai) [reply examples/twilio_api_v2010.yaml:244]: This is needed since conditional attribute is a list of list, and we need to pick any one of the properties from each list.
so for example we can have the properties like this 
`{
  "x-twilio": {
    "conditional": [
      [
        "url",
        "twiml",
        "application_sid"
      ],
	[
        "
        “Foo”
	“bar”
      ]
    ]
  }
}`

representing this would require double dashes in yaml

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 2 above)
Comment (kridai) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java:102]: This is needed since for we are representing "ACCOUNT_SID" as "SID" for some of the apis , for example in https://github.com/twilio/twilio-oai/blob/main/spec/yaml/twilio_api_v2010.yaml#L27129-L27135 we are calling ACCOUNT_SID as SID so we can no longer rely upon the paramName, instead we are checking for the pattern

**Review by childish-sambino — APPROVED** [review]
